src/lib/kdf/kdf2/kdf2.cpp

```diff
@@ -11,7 +11,8 @@ namespace Botan {
 
 size_t KDF2::kdf(byte key[], size_t key_len,
                  const byte secret[], size_t secret_len,
-                 const byte salt[], size_t salt_len) const
+                 const byte salt[], size_t salt_len,
+                 const byte[], size_t) const
```

Comment:
I understand your concern, but wouldn't it be better to throw an exception in case label is set for a KDF that does not use one?